### ISSUE TITLE
upload gives relative/absolute path error

### ISSUE THREAD

**Issue by satra** [dandi/dandi-cli#52]
```
$ dandi upload -d 000001
2020-03-14 21:53:02,654 [    INFO] Found 136 files to consider
Error: Both paths must either be absolute or relative. Got '/net/vast-storage.ib.cluster/scratch/Mon/satra/000001/dandiset.yaml' and '000001'
```

**Comment by yarikoptic** [comment]
fun.  so -- are you are outside of `000001`, and inside of it -- do you have symlinks or copied data? are symlinks relative?

also if you rerun with `--pdb` and post traceback - would be appreciated

I have tried to replicate to just see that `-d` seems to lead to all kinds of incorrect treatment.
You should be able to just cd to your dandiset and run `dandi upload`

**Comment by yarikoptic** [comment]
the gotchas I ran into:
```
lena:~/.tmp/dandiset-D5M6e1G/organized
$> DANDI_DEVEL=1 dandi upload -i local-docker -d organized       
2020-03-14 22:01:21,234 [    INFO] Found 2 files to consider
PATH                 SIZE      ERRORS UPLOAD STATUS            MESSAGE                                           
dandiset.yaml        369 Bytes   0           ERROR             Write access denied for collection 5e6ab9e6ed07...
sub-1/sub-1.nwb      792.8 kB    0           ERROR             Write access denied for collection 5e6ab9e6ed07...
Summary:             793.2 kB                2 ERROR           2 Write access denied for collection 5e6ab9e6ed...
```
and I am already in `organized` -- should have just crashed since there is no organized subdir

and with attempt to upload from outside of it:
```
lena:~/.tmp/dandiset-D5M6e1G
$> DANDI_DEVEL=1 dandi upload -i local-docker -d organized
Traceback (most recent call last):
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/bin/dandi", line 11, in <module>
    load_entry_point('dandi', 'console_scripts', 'dandi')()
  File "/usr/lib/python3/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/yoh/proj/dandi/dandi-cli/dandi/cli/command.py", line 118, in wrapper
    return f(*args, **kwargs)
  File "/home/yoh/proj/dandi/dandi-cli/dandi/cli/cmd_upload.py", line 104, in upload
    f"Found no {dandiset_metadata_file} anywhere.  Use 'dandi register', 'download', or 'organize' first"
RuntimeError: Found no dandiset.yaml anywhere.  Use 'dandi register', 'download', or 'organize' first
```

**Comment by yarikoptic** [comment]
ok, for now I think I will just hide `-d` for upload, and reschedule the fix for 0.4.2

**Comment by yarikoptic** [comment]
`-d` was completely removed some time before 0.6.0, so I will just close for now
